roles/ceph-osd/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: install dependencies
+  apt: pkg=parted
+       state=present
+
+- name: write out bootstrap keyring
+  copy:
+    dest: "{{ hostvars[groups['mons'][0]]['keyrings']['source'] }}"
+    content: "{{ hostvars[groups['mons'][0]]['keyrings']['content'] | b64decode }}"
+
+- name: prepare osd disk(s)
+  command: "ceph-disk prepare {{ item.0 }} {{ item.1 }}"
+  with_together:
+    - ceph_osd.devices
+    - ceph_osd.raw_journal_devices
+  changed_when: false
+  failed_when: false
```

Comment:
Might also deserve a comment here as to why failure is masked. What happens if there is a real failure?